csvcubed/csvcubed/cli/inspect/metadataprinter.py

```diff
@@ -87,11 +105,14 @@ def gen_dsd_info_printable(self) -> str:
         result_dataset_label_dsd_uri: DSDLabelURIResult = (
             select_csvw_dsd_dataset_label_and_dsd_def_uri(self.csvw_metadata_rdf_graph)
         )
-        self.dsd_uri = result_dataset_label_dsd_uri.dsd_uri
+        self.dsd_uri: str = result_dataset_label_dsd_uri.dsd_uri
 
         result_qube_components: QubeComponentsResult = select_csvw_dsd_qube_components(
             self.csvw_metadata_rdf_graph, self.dsd_uri, self.csvw_metadata_json_path
         )
+        self.qube_components: List[
```

Comment:
If you do turn this into a `@dataclass`, you should still list this propety but you do the following:

```python
qube_components: List[QubeComponentResult] = field(init=False)
```

Reply:
Infact much of this could go into the `__post_init__` function of a dataclass.

https://docs.python.org/3/library/dataclasses.html#post-init-processing